pennylane/compiler/qjit_api.py

```diff
@@ -621,7 +622,7 @@ def _get_for_loop_qfunc_prim():
 
     import jax  # pylint: disable=import-outside-toplevel
 
-    for_loop_prim = jax.core.Primitive("for_loop")
+    for_loop_prim = create_non_jvp_primitive()("for_loop")
```

Comment:
Don't we need this for `while_loop` as well?

Reply:
Good catch, thanks! Done.